src/main/java/cafe/cryptography/curve25519/EdwardsPoint.java

```diff
@@ -263,7 +263,38 @@ public EdwardsPoint multiply(final Scalar s) {
      */
     public static EdwardsPoint vartimeDoubleScalarMultiplyBasepoint(final Scalar a, final EdwardsPoint A,
             final Scalar b) {
-        throw new UnsupportedOperationException();
+        final byte[] aNaf = a.nonAdjacentForm();
+        final byte[] bNaf = b.nonAdjacentForm();
+
+        ProjectiveNielsPoint.NafLookupTable tableA = ProjectiveNielsPoint.buildNafLookupTable(A);
```

Comment:
In `ed25519-java` this was pre-computed when `A` was negated outside the function, as this gets used in the verification equation, and it's possible that many signatures might be verified with the same public key while it is in-memory. For now we compute the table here; later on, when the lookup table usability is improved, we could allow users to decide whether to pre-compute or not (probably by having this function take the lookup table directly as an argument instead of the `EdwardsPoint`).